staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -916,15 +901,17 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 		controller, controllerCompletedCh := startAPFController(t, stopCh, apfConfiguration, serverConcurrency, plName, plConcurrency)
 
 		headerMatcher := headerMatcher{}
-		rquestTimesOutPath := "/request/time-out-as-designed"
+		rquestTimesOutPath := "/request/time-out-as-expected"
 		reqHandlerErrCh, callerRoundTripDoneCh := make(chan error, 1), make(chan struct{})
 		requestHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			headerMatcher.inspect(t, w, fsName, plName)
 
 			if r.URL.Path == rquestTimesOutPath {
-
-				// inner handler writes header and then let the request time out.
+				// after the inner handler writes, we let the request time out.
 				w.WriteHeader(http.StatusBadRequest)
+				if _, err := w.Write([]byte("hello world")); err != nil {
```

Comment:
why was this change made?

Reply:
the inner handler should write to the underlying `ResponseWriter` object before the request times out, I think this is the original intention of the test, given from its name.

`Write` is more interesting to us  than `WriteHeader` in relation to the timeout filter and request deadline. 
https://github.com/kubernetes/kubernetes/blob/5837de2796d6b0c17f0d26ecf16f24ca321964dc/staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go#L196-L202

I am trying to document these behaviors in this PR https://github.com/kubernetes/kubernetes/pull/124730 as a pre-requisite for https://github.com/kubernetes/enhancements/pull/4462

Let me know your thoughts